src/components/campaigns/grid/EditForm.tsx

```diff
@@ -64,15 +66,42 @@ const validationSchema: yup.SchemaOf<Omit<CampaignEditFormData, 'campaignFiles'>
       .transform(parseDateString)
       .min(yup.ref('startDate'), `end date can't be before start date`),
     state: yup.mixed().oneOf(Object.values(CampaignState)).required(),
+    currency: yup.mixed().oneOf(Object.values(Currency)).required(),
   })
 
-export default function EditForm({ campaign }: { campaign: CampaignResponse }) {
+const statesForDisableCurrencySelect = [
+  CampaignState.rejected,
+  CampaignState.suspended,
+  CampaignState.complete,
+  CampaignState.disabled,
+  CampaignState.error,
+  CampaignState.deleted,
+]
+
+export default function EditForm({ campaign }: { campaign: AdminSingleCampaignResponse }) {
   const router = useRouter()
   const queryClient = useQueryClient()
   const [files, setFiles] = useState<File[]>([])
   const [roles, setRoles] = useState<FileRole[]>([])
   const { t } = useTranslation()
 
+  const incomingTransfersAmount = useMemo(() => {
+    return campaign.incomingTransfers.reduce((acc, transfer) => {
+      return acc + transfer.amount
+    }, 0)
+  }, [campaign])
+
+  const donationsAmount = useMemo(() => {
+    return campaign.vaults.reduce((acc, vault) => {
+      return acc + vault.amount
+    }, 0)
+  }, [campaign])
+
+  const IsCurrencySelectDisabled =
+    statesForDisableCurrencySelect.includes(campaign.state) ||
+    (campaign.state === CampaignState.active &&
```

Comment:
please move the CampaignState.active to the statesForDisableCurrencySelect too, so this condition to become simpler.

Reply:
Income value > 0 is a second and compulsory condition only for  `active` state:
```
const IsCurrencySelectDisabled =
    statesForDisableCurrencySelect.includes(campaign.state) && (incomingTransfersAmount > 0 || donationsAmount > 0)
```
`active` && income = 0  => false /correct/
`active` && income = 100 => true /correct/
`rejected`/`error` && income = 0 (it is supposed their income to be 0) => false /incorrect/

I am not sure how to optimize the function.